Mathlib/Data/Int/Cast/Lemmas.lean

```diff
@@ -220,6 +220,11 @@ theorem eq_intCast' [AddGroupWithOne α] [FunLike F ℤ α] [AddMonoidHomClass F
     ∀ n : ℤ, f n = n :=
   DFunLike.ext_iff.1 <| (f : ℤ →+ α).eq_intCastAddHom h₁
 
+/-- This version is primed so that the `RingHomClass` versions aren't. -/
+theorem map_intCast' [AddGroupWithOne α] [AddGroupWithOne β] [FunLike F α β]
```

Comment:
Is this not a strict generalization?

Reply:
Do you mean `map_intCast`? That one assumes RingHomClass so there's no need to supply a proof of `f 1 = 1`. This new lemma is in parallel to [map_natCast'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Cast/Basic.html#map_natCast').